tests/integration/test_cli_service_accounts.py

```diff
@@ -0,0 +1,610 @@
+    )
+
+    # Check if service account has been deleted
+    service_account_result = subprocess.run(
+        ["kubectl", "get", "serviceaccount", username, "-n", namespace, "-o", "json"],
+        capture_output=True,
+        text=True,
+    )
+    assert service_account_result.returncode != 0
+
+    # Check if the role corresponding to the service account has also been deleted
+    role_result = subprocess.run(
+        ["kubectl", "get", "role", role_name, "-n", namespace, "-o", "json"],
+        capture_output=True,
+        text=True,
+    )
+    assert role_result.returncode != 0
+
+    # Check if the associated role binding has been deleted as well
+    role_binding_result = subprocess.run(
+        [
+            "kubectl",
+            "get",
+            "rolebinding",
+            role_binding_name,
+            "-n",
+            namespace,
+            "-o",
+            "json",
+        ],
+        capture_output=True,
+        text=True,
+    )
+    assert role_binding_result.returncode != 0
+
+    # Check for RBAC permissions, these should be invalid now
+    sa_identifier = f"system:serviceaccount:{namespace}:{username}"
+    rbac_check = subprocess.run(
+        [
+            "kubectl",
+            "auth",
+            "can-i",
+            action,
+            resource,
+            "--namespace",
+            namespace,
+            "--as",
+            sa_identifier,
+        ],
+        capture_output=True,
+        text=True,
+    )
+    assert rbac_check.returncode != 0
+    assert rbac_check.stdout.strip() == "no"
+
+
+@pytest.mark.parametrize("backend", VALID_BACKENDS)
+def test_delete_service_account_that_does_not_exist(namespace, backend):
+    username = str(uuid.uuid4())
+
+    stdout, stderr, ret_code = run_service_account_registry(
+        "delete", "--username", username, "--namespace", namespace, "--backend", backend
+    )
+    assert ret_code != 0
+    assert stdout.strip() == f"Account {username} could not be found."
+
+
+@pytest.mark.parametrize("backend", VALID_BACKENDS)
+def test_service_account_get_primary(namespace, backend):
+    """Test retrieval of primary service account using the CLI.
+
+    Creates an service account with --primary option provided,
+    and then checks whether the same service account is returned
+    upon calling `get-primary` sub-command.
+    """
+    username = str(uuid.uuid4())
+
+    # Create a new service account with --primary option provided
+    run_service_account_registry(
+        "create",
+        "--username",
+        username,
+        "--namespace",
+        namespace,
+        "--backend",
+        backend,
+        "--primary",
+    )
+
+    # Attempt to get primary service account, this should be the same
+    # as the one created before
+    stdout, stderr, ret_code = run_service_account_registry(
+        "get-primary", "--backend", backend
+    )
+    assert f"{namespace}:{username}" == stdout.strip()
```

Comment:
Can you also check that the output `list` command is correct?

Reply:
Updated.